src/common.mak

```diff
@@ -0,0 +1,204 @@
+# Define library files and import paths
+################################################################################
+
+DRUNTIME_IMPORT_DIR=$(DRUNTIME_DIR)/import
+PHOBOS_IMPORT_DIR=$(PHOBOS_DIR)
+
+ifneq (,$(DRUNTIME))
+	CUSTOM_DRUNTIME=1
+endif
+
+ifeq (,$(findstring win,$(OS)))
+   LIB_DRUNTIME = $(DRUNTIME_BUILD_DIR)/libdruntime.a
+   LIB_DRUNTIMESO = $(basename $(LIB_DRUNTIME)).so.a
+else
+   LIB_DRUNTIME = $(DRUNTIME_PATH)/lib/druntime.lib
+endif
+
+# Set PHOBOS name and full path
+ifeq (,$(findstring win,$(OS)))
+	LIB_PHOBOS = $(PHOBOS_BUILD_DIR)/libphobos2.a
+	LIB_PHOBOSSO = $(PHOBOS_BUILD_DIR)/libphobos2.so
+endif
+
+# build with shared library support
+# (defaults to true on supported platforms, can be overridden w/ make SHARED=0)
+SHARED=$(if $(findstring $(OS),linux freebsd),1,)
+LINKDL=$(if $(findstring $(OS),linux),-L-ldl,)
+
+################################################################################
+# Set CC and DMD
+################################################################################
+
+ifeq ($(OS),win32wine)
+	CC = wine dmc.exe
+	DMD = wine dmd.exe
+	RUN = wine
+else
+	DMD = $(DMD_BUILD_DIR)/dmd
+	ifeq ($(OS),win32)
+		CC = dmc
+	else
+		CC = cc
+	endif
+	RUN =
+endif
+
+################################################################################
+# Commonly used binaries
+################################################################################
+
+DUB=dub
+
+################################################################################
+# Platform-specific variables
+################################################################################
+
+# Set DOTOBJ and DOTEXE
+ifeq (,$(findstring win,$(OS)))
+	DOTOBJ:=.o
+	DOTEXE:=
+	PATHSEP:=/
+else
+	DOTOBJ:=.obj
+	DOTEXE:=.exe
+	PATHSEP:=$(shell echo "\\")
+endif
+
+ifeq (osx,$(OS))
+	DOTDLL:=.dylib
+	DOTLIB:=.a
+else
+	DOTDLL:=.so
+	DOTLIB:=.a
+endif
+
+################################################################################
+# Phobos DFLAGS (for linking with the currently built Phobos)
+################################################################################
+
+PHOBOS_DFLAGS=-conf= $(MODEL_FLAG) -I$(DRUNTIME_IMPORT_DIR) -I$(PHOBOS_IMPORT_DIR) -L-L$(PHOBOS_BUILD_DIR) $(PIC)
+ifeq (1,$(SHARED))
+PHOBOS_DFLAGS+=-defaultlib=libphobos2.so -L-rpath=$(PHOBOS_BUILD_DIR)
+endif
+
+################################################################################
+# Default DUB FLAGS
+################################################################################
+
+DUBFLAGS = --arch=$(subst 32,x86,$(subst 64,x86_64,$(MODEL)))
+
+################################################################################
+# Automatically create dlang/tools repository if non-existent
+################################################################################
+
+_all: all
+
+$(TOOLS_DIR):
+	git clone --depth=1 ${GIT_HOME}/$(@F) $@
+
+$(TOOLS_DIR)/checkwhitespace.d: | $(TOOLS_DIR)
```

Comment:
This section is optional and if people are against it I could remove it, but the fact is that we already have this rule in almost all Makefile, so this stops the redundancy.

Reply:
There are other target which are common too, e.g. `detab` or `gitzip`, but I wasn't sure whether they are really still in use.